### ISSUE TITLE
Lost the ability to read temperature

### ISSUE THREAD

**Issue by jul9000** [snOOrz/homebridge-aqara#27]
Hi,

I was successfully using HomeBridge with both yeelight and aqara plugins. I had no issues at all until today, when I've decided to change the location of my Xiaomi Temperature and Humidity Sensor. I also wanted to clear the temperature history within the Xiaomi Home app, so I decided to remove the device there. After doing this, I have paired it again. Unfortunately in the iOS Home app the sensor doesn't show the temperature anymore! The weird thing is that the humidity is still there, but the temperature "box" says "No connection".

I have already recreated the whole homebridge setup:
1. removed all the files in /accessories
2. removed all the files in /persist
3. changed the name of the bridge in config.json
The sensor showed up, shows humidity without problems, but not the temperature.

Another weird thing, is that the Xiaomi Home app shows both the humidity and temperature just fine (so the sensor itself isn't damaged).

What's wrong? Can you help me?

**Comment by snOOrz** [comment]
You can remove homebridge from iOS Home app, and add it again.

On 2017年1月6日 +0800 04:35, jul9000 <notifications@github.com>, wrote:

Hi,

I was successfully using HomeBridge with both yeelight and aqara plugins. I had no issues at all until today, when I've decided to change the location of my Xiaomi Temperature and Humidity Sensor. I also wanted to clear the temperature history within the Xiaomi Home app, so I decided to remove the device there. After doing this, I have paired it again. Unfortunately in the iOS Home app the sensor doesn't show the temperature anymore! The weird thing is that the humidity is still there, but the temperature "box" says "No connection".

I have already recreated the whole homebridge setup:

  1.  removed all the files in /accessories
  2.  removed all the files in /persist
  3.  changed the name of the bridge in config.json
The sensor showed up, shows humidity without problems, but not the temperature.

Another weird thing, is that the Xiaomi Home app shows both the humidity and temperature just fine (so the sensor itself isn't damaged).

What's wrong? Can you help me?

**Comment by jul9000** [comment]
I've already tried that multiple times, unfortunately without success:

- tried to remove only the gateway,
- tried to remove the entire home,
- tried to reboot the computer hosting homebridge
- tired to reboot my iphone

What else can I try? Is there a way to access some kind of logs? I've tried to run "homebridge -D", but nothing really happens...

**Comment by tru2dagame** [comment]
I got two Temperature and Humidity Sensor. Always got issue with one temperature information. And the other one works well.

**Comment by jul9000** [comment]
Worst thing is that I had it working just fine, but suddenly it stopped working.

I've also tried to unpair and pair again the sensor with the gateway, using the Xiaomi Home app. The readings shows up just fine there, only the Apple Home app can't get the temperature. I have to say, it's frustrating ;)

It looks like homebridge is able to connect, but doesn't get the correct reading:
- when homebridge is running the sensor status says: "no answer"
- but when homebridge isn't running the sensor status says: "unavailable"

**Comment by snOOrz** [comment]
I have no idea.
As long as the device works in Mijia app, it should work in homebridge too.


On 2017年1月6日 +0800 18:27, jul9000 <notifications@github.com>, wrote:

Worst thing is that I had it working just fine, but suddenly it stopped working.

I've also tried to unpair and pair again the sensor with the gateway, using the Xiaomi Home app. The readings shows up just fine there, only the Apple Home app can't get the temperature. I have to say, it's frustrating ;)

It looks like homebridge is able to connect, but doesn't get the correct reading:

  *   when homebridge is running the sensor status says: "no answer"
  *   but when homebridge isn't running the sensor status says: "unavailable"

**Comment by jul9000** [comment]
Is it possible that negative readings causes this issue? It's quite cold right now in Poland (-12C). I've noticed similar problems in other homebridge plugins.

**Comment by marcotini** [comment]
You are not alone, also my temperature sensor doesn't working... just the humidity.

**Comment by jul9000** [comment]
Yep, you can't rely on them too much. Mine was suddenly showing reading yesterday for a couple of hours, then disappeared and reappeared today... I'm wondering what's causing this? Isn't there any way to debug?

**Comment by tru2dagame** [comment]
After missing for one day. It's now back to be normal for two days. I didn't do any changes. 

Best Regards,
Tru

**Comment by marcotini** [comment]
Even to me now it's magically work (the only thing I did is unplug and replug the gateway).

**Comment by marcotini** [comment]
I found the solution: you need to deselect the option of status (so you don't see the temperature in homepage but just in the right room where the sensor is place).

**Comment by jul9000** [comment]
Unfortunately, I cannot confirm this solution.
